controllers/dnspolicy_controller.go

```diff
@@ -199,5 +202,35 @@ func (r *DNSPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				return gatewayEventMapper.MapToPolicy(ctx, object, &v1alpha1.DNSPolicy{})
 			}),
 		)
+	r.RegisterEventHandler(&DNSMetricsEventHandler{})
 	return ctrlr.Complete(r)
 }
+
+const (
+	dnsRecordNameLabel      = "dns_record_name"
+	dnsRecordNamespaceLabel = "dns_record_namespace"
+	dnsRecordEventTypeLabel = "dns_record_event_type"
+	dnsPolicyNameLabel      = "dns_policy_name"
+	dnsPolicyNamespaceLabel = "dns_policy_namespace"
+	dnsPolicyReadyLabel     = "dns_policy_ready"
+)
+
+var (
+	dnsRecordEvent = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "dns_record_event",
```

Comment:
is the naming here consistent with other metrics? I think we should prob have a naming convention for metrics we emit WDYT

Reply:
as far as I can tell, these are the only 2 metrics in the kuadrant-operator right now?

Reply:
There's a good guide on naming metrics at https://prometheus.io/docs/practices/naming/